### PR TITLE
fix(TDI-48213):fix the risk of not closing resources

### PR DESCRIPTION
**Pull request by vyu-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-48213

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-48213

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by pyzhou-talend — APPROVED** [review]
LGTM

File: main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_finally.javajet [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+%>
+Object filename_<%=cid %> = <%=filename %>;
+if(!(filename_<%=cid %> instanceof java.io.InputStream)){
+
+	org.talend.fileprocess.delimited.RowParser reader_<%=cid %> = (org.talend.fileprocess.delimited.RowParser)resourceMap.get("reader_<%=cid %>"); 
+    org.talend.fileprocess.delimited.RowParserByByte byteReader_<%=cid %> = (org.talend.fileprocess.delimited.RowParserByByte)resourceMap.get("byteReader_<%=cid %>"); 
+	if(reader_<%=cid %>!=null) {
+		try {
+			reader_<%=cid %>.close();
+		}catch(java.lang.Exception e_reader_<%=cid%>) {
```
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_finally.javajet:25]: Do we really need java.lang.Exception here? ：）

File: main/plugins/org.talend.designer.components.localprovider/components/tFileOutputDelimited/tFileOutputDelimited_begin.javajet [hunk 2]
```diff
@@ -487,7 +487,9 @@ if(("false").equals(ElementParameterParser.getValue(node,"__CSV_OPTION__"))) {
                         new java.io.FileOutputStream(fullName_<%=cid%> + splitedFileNo_<%=cid%> + extension_<%=cid%>, <%=isAppend%>),<%=encoding%>));
                 <%
                 }
-
+                %>
+                resourceMap.put("out_<%=cid%>", out<%=cid%>);
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileOutputDelimited/tFileOutputDelimited_begin.javajet:491]: Are we missing a finally file for tFileOutputDelimited that guarantees the closure of out_?

Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tFileOutputDelimited/tFileOutputDelimited_begin.javajet:491]: The closure of out_ has already been involved in [tFileOutputDelimited_finally.javajet](https://github.com/Talend/tdi-studio-se/blob/7d79ca27badec5520a8f34d46d1f6128e70822f9/main/plugins/org.talend.designer.components.localprovider/components/tFileOutputDelimited/tFileOutputDelimited_finally.javajet#L67-L68).

File: main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_finally.javajet (hunk 1 above)
Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_finally.javajet]: Discussed with @jzhao-talend, that the try/catch is not needed, so removed.

**Review by coheigea — DISMISSED** [review]

File: main/plugins/org.talend.designer.components.localprovider/components/tFileInputExcel/tFileInputExcel_begin.javajet [hunk 3]
```diff
@@ -54,6 +54,9 @@ if ((metadatas!=null)&&(metadatas.size()>0)) {
 
 		String mode = ElementParameterParser.getValue(node,"__GENERATION_MODE__");
 		final boolean eventMode = "EVENT_MODE".equals(mode);
+		
+		boolean configureRatio = ("true").equals(ElementParameterParser.getValue(node,"__CONFIGURE_COMPRESSION_RATIO__"));
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileInputExcel/tFileInputExcel_begin.javajet:58]: please revert the wrong commit.

**Review by jzhao-talend — CHANGES_REQUESTED** [review]

File: main/plugins/org.talend.designer.components.localprovider/components/tFileInputExcel/tFileInputExcel_begin.javajet (hunk 3 above)
Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tFileInputExcel/tFileInputExcel_begin.javajet:58]: @jzhao-talend Will do it, many thanks!

File: main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_end.javajet [hunk 4]
```diff
@@ -26,16 +26,6 @@
 
 			<%log4jFileUtil.retrievedDataNumberInfo(node);%>
 			
-if(!(filename_<%=cid %> instanceof java.io.InputStream)){
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_end.javajet:29]: why remove this part of code ?

Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tFileInputPositional/tFileInputPositional_end.javajet:29]: this part of code is moved to finally part

**Review by jzhao-talend — APPROVED** [review]
LGTM

**Pull request merged**